### DISCUSSION TITLE
Creating a Pull Request - LGMSOC Participant(Anshika)

### DISCUSSION THREAD

**Discussion by prathimacode-hub** [prathimacode-hub/Awesome_Python_Scripts#213] [Q&A]
I shall share you the easiest way to creat a pull request without any hassles especially for beginners to open source and github

**Comment by AK-Subodh** [comment]
> 
> 
> I shall share you the easiest way to creat a pull request without any hassles especially for beginners to open source and github

OK Mam!

**Comment by prathimacode-hub** [comment]
Process to Follow (without command line confusion)👍

Fork the repository. You are now in your profile repository.
You can see a "Add File" option in the local repository
You can add files or folders using the 2options given
Create new file option helps in creating both file and sub repository within this repository.
Example : You want to add up your "Oddly Satisfying Game" Script to the repository
Go to "PyGamesScripts" repository. Select "Create a new file" option,
It opens up like this - /Awesome_Python_Scripts/PyGamesScripts
To add up your repository and file you do :
/Awesome_Python_Scripts/PyGamesScripts/Oddly Satisfying Game/README.md
This add up would not only add 'Oddly Satisfying Game' as repository but also adds a empty 'README.md' file.
You can add your README content in 'README.md file' and commit the changes
Now you have created a repository of your desired script, you are yet to add your python program and related files.
From the very same "Add File" option, select "Upload files" and add your python program from your local drive of your system.
Make sure the script file name is oddly_satisfying_game.py before you upload the file to this repository.
Now, we needed to add images and screenshots of your project, so we again use " Create a New File" Option you do
/Awesome_Python_Scripts/PyGamesScripts/Oddly Satisfying Game/Images/README.md and commit it
You're now in "Images" folder. Add the relevant images and screenshots you want to add up in your README.md file of your project using "Add File" Option from local drive of your system
Get back to your project repository of 'Oddly Satisfying Game' and using "Create a New File" Option you do
/Awesome_Python_Scripts/PyGamesScripts/Oddly Satisfying Game/requirements.txt, add the imported libraries used in your program in this file and commit it
We haven't added our output screenshots or images in "README.md" of your project. Open up your README file. There you could see a "pencil button" that is to edit the file. You can add up the image links you had stored in "Images" folder in README file
If we remember, we had created a README.md file in "Images" folder as well to get it registered as a repository.
You can now delete that file using "delete option that is trash button". And you are done.
Finally, You have successfully added up all your required files needed.
Now select "Contribute" option above and select Open Pull request, and add a defined title of your project, a proper commit message in a description using the PR template given.
Finally, "Create a Pull Request" option to synchronize your fork to the committed repository. That's it.
Voila, you did it. Congrats, on learning how to create a PR in a easier way. No much fuss around to deal with.
PR can be created without the use of command line terminal, but preferrably that way to show professionalism.
Contribution is all that matters, so start contributing and feel free to do more PR in future.

@AK-Subodh , I like your confidence and enthusiasm to learn. Keep going.
If you found this helpful and easier to try, you can do create the PR of the same script with required changes I have mentioned in this comment.

Looking forward to merge the upcoming PR from you. Thank You.

**Reply by AK-Subodh** [reply]
@prathimacode-hub 
Mam, I forked the repository. Now as you said, I'm creating the empty README file. At this point there are two commit options, default is "Commit directly to main". So should I create a branch here or commit to main ?
Also, I previously forked the repository and added all the files under same name as "Oddly-Satisfying-Game". They are still there. What should I do about them?

**Comment by prathimacode-hub** [comment]
Commit to main here. This is forked repsitory. When you commit to mine it should be in branch. No, you save those files and delete that old ones. And do it again then it won't be repeated. @AK-Subodh

**Comment by prathimacode-hub** [comment]
You should do this method from scratch. First save your files, delete the forked repository. Refork it again and do this process. For that I have to close this PR, so that your changes won't come to same PR. I shall close your PR, and you do it again. Remember to save your files, before you do.
@AK-Subodh

**Reply by AK-Subodh** [reply]
@prathimacode-hub Mam, I created the pull request, Please review it. Thank you!

**Reply by prathimacode-hub** [reply]
> @prathimacode-hub Mam, I created the pull request, Please review it. Thank you!

Reviewed it. Kindly check. @AK-Subodh
